src/k8s/pkg/k8s/client/cluster.go

```diff
@@ -22,21 +20,7 @@ func (c *k8sdClient) IsBootstrapped(ctx context.Context) bool {
 }
 
 // Bootstrap bootstraps the k8s cluster
-func (c *k8sdClient) Bootstrap(ctx context.Context, bootstrapConfig apiv1.BootstrapConfig) (apiv1.NodeStatus, error) {
-	// Get system hostname.
-	rawHostname, err := os.Hostname()
-	if err != nil {
-		return apiv1.NodeStatus{}, fmt.Errorf("failed to retrieve system hostname: %w", err)
-	}
-	// TODO: this should be done on the server side, but we cannot currently hijack the microcluster bootstrap endpoint.
-	hostname, err := utils.CleanHostname(rawHostname)
-	if err != nil {
-		return apiv1.NodeStatus{}, fmt.Errorf("invalid hostname %q: %w", rawHostname, err)
-	}
-
-	// Get system addrPort.
-	addrPort := util.CanonicalNetworkAddress(util.NetworkInterfaceAddress(), config.DefaultPort)
-
+func (c *k8sdClient) Bootstrap(ctx context.Context, hostname string, address string, bootstrapConfig apiv1.BootstrapConfig) (apiv1.NodeStatus, error) {
```

Comment:
Maybe just add those new parameters to the `BootstrapConfig`?

Reply:
hmm, the `BootstrapConfig` is passed verbatim to the bootstrap hook, whereas `name` and `address` here are explicitly needed by microcluster.

It would look someting like:

```
NewCluster(BootstrapConfig.Name, BootstrapConfig.Address, BootstrapConfig)
```

which is not very clean

Reply:
fair - let's keep it.